tests/e2e/support/page/files/publicLink.ts

```diff
@@ -0,0 +1,227 @@
+  }
+
+  getDateType = (expiryDate: string): string => {
+    if (expiryDate.charAt(0).includes('-')) {
+      throw new Error('The provided date is negative and has already expired !!')
+    } else if (expiryDate.charAt(0).includes('+')) {
+      return expiryDate.toLowerCase().match(/[dayrmonthwek]+/)[0]
+    }
+  }
+
+  addMonth = (noOfMonths: number): Date => {
+    const date = new Date()
+    date.setMonth(date.getMonth() + noOfMonths)
+    return date
+  }
+
+  addDays = (days: number): Date => {
+    const date = new Date()
+    date.setDate(date.getDate() + days)
+    return date
+  }
+
+  setActualExpiryDate = (dateType: string, dateOfExpiration: string): Date => {
+    switch (dateType) {
+      case 'day':
+        return this.addDays(parseInt(dateOfExpiration))
+      case 'week':
+        return this.addDays(parseInt(dateOfExpiration) * 7)
+      case 'month':
+        return this.addMonth(parseInt(dateOfExpiration))
+      case 'year':
+        return new Date(
+          new Date().setFullYear(new Date().getFullYear() + parseInt(dateOfExpiration))
+        )
+    }
+  }
+
+  async selectDate(dateOfExpiration: string): Promise<void> {
+    const newExpiryDate = this.setActualExpiryDate(this.dateType, dateOfExpiration)
+    const expiryDay = newExpiryDate.getDate()
+    const expiryMonth = ('0' + (newExpiryDate.getMonth() + 1)).slice(-2)
+    const expiryYear = newExpiryDate.getFullYear().toString()
+    await this.expirationDateDropdownLocator.click()
+    const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']
+    const months = [
+      'January',
+      'February',
+      'March',
+      'April',
+      'May',
+      'June',
+      'July',
+      'August',
+      'September',
+      'October',
+      'November',
+      'December'
+    ]
+    const dayMonthYear =
+      days[newExpiryDate.getDay()] +
+      ', ' +
+      months[newExpiryDate.getMonth()] +
+      ' ' +
+      expiryDay +
+      ', ' +
+      expiryYear
+    await this.monthAndYearDropdownLocator.click()
+    if ((await this.yearButtonLocator.innerText()) !== expiryYear) {
+      await this.yearButtonLocator.click()
+      while (true) {
+        const nextYearSpanValue = await this.yearButtonLocator.innerText()
+        const splitNextSpanYear = nextYearSpanValue.split('-')
+        if (
+          newExpiryDate.getFullYear() >= parseInt(splitNextSpanYear[0]) &&
+          newExpiryDate.getFullYear() <= parseInt(splitNextSpanYear[1])
+        ) {
+          const yearLocator = await this.actor.page.locator(
+            util.format(this.yearSelector, expiryYear)
+          )
+          await yearLocator.click()
+          break
+        }
+        await this.nextSpanYearLocator.click()
+      }
+    }
+
+    await this.selectExpiryMonth(expiryYear, expiryMonth)
+    await this.selectExpiryDay(dayMonthYear)
+  }
+
+  async isPublicLinkCreated(noOfPublicLink: number): Promise<void> {
```

Comment:
```suggestion
  async getLinksCount(noOfPublicLink: number): Promise<void> {
```
it's no good to perform assertions inside page object methods. just get or set page information from the PO methods and do your assertions in the context.